byway/src/main/webapp/destinations.js

```diff
@@ -133,17 +129,52 @@ function addMarker(searchBox,map){
 * fetches start location and destinations from DestinationsServlet and adds to DOM
 */
 function updateLocations(locationData){
-    document.getElementById('start-location').innerText = "Start Location :" + locationData.start;
     const container = document.getElementById('destinations-container');
-    container.innerText = "Destinations:";
+    container.innerText = "";
     let destinationArray= locationData.destinations;
     destinationArray.forEach((destination) => {
-      let destinationToAdd = document.createElement('p');
-      destinationToAdd.innerText = destination;
-      container.appendChild(destinationToAdd);
+      const request = {
+        query: destination,
+        fields: ["name", "photos", "formatted_address", "rating", "business_status"]
+      };
+      let service = new google.maps.places.PlacesService(map);
+      service.findPlaceFromQuery(request, (results, status) => {
+        if (status === google.maps.places.PlacesServiceStatus.OK) {
+          addLocationtoDOM(results, container);
+        }
+        else{
+          let destinationToAdd = document.createElement('p');
```

Comment:
So what is the logic here if the places service doesn't return an expected result? Only the place name is shown? That seems like a confusing experience for the user. Also, what happens if one of these invalid places is passed to another page or stored on the trip entity? This could lead to some weird behavior. It would be better to handle the failure in a more obvious way to the user. You could have an error message on the page saying the places are invalid, or you could display them in a separate section in red as unavailable, etc

Reply:
Ok I changed it to an alert instead